src/box/lua/config/applier/lua.lua

```diff
@@ -0,0 +1,37 @@
+local alloc = require('internal.alloc')
+local log = require('internal.config.utils.log')
+
+local is_lua_memory_applied = false
+
+local function apply(config)
+    local configdata = config._configdata
+    local memory = configdata:get('lua.memory', {use_default = true})
+    local old_memory = alloc.getmemory()
+
+    -- If memory limit is applied for the first time or it is
+    -- increased.
+    if not is_lua_memory_applied or memory > old_memory then
+        -- The call throws an error if the user tries to limit the
+        -- memory with values < 256MB or with an amount less than
+        -- currently allocated.
+        alloc.setmemory(memory)
+
+        log.verbose(('lua.apply: set memory limit to %d'):format(memory))
+    -- If memory limit is decreased.
+    elseif memory < old_memory then
+        local name = 'lua_memory_decreased'
+        local warning = 'lua.apply: lua.memory decrease will not be applied ' ..
+                        'until the instance is restarted'
+
+        config._aboard:set({type = 'warn', message = warning}, {key = name})
+    end
+    -- Nothing to do if memory limit is unchanged.
+
+
+    is_lua_memory_applied = true
```

Comment:
I think that maybe we should alow to decrease the memory limit, if it is not less than the currently allocated memory amount, since I see no drawbacks of this. Also, this alows to drop the special first-call case.

But I'm OK with your's or Georgiy's approach.

Reply:
I've worked on this and I suggest to allow the limit to be decreased up to the current amount of the used Lua memory plus some extra unused space (I've chosen arbitrary value of 16MB) to make sure not to instantly throw and error after setting the limit. If there is already too much allocated memory the alert is issued.

Reply:
Allowed limit values in `[used+16MiB; +inf]` look OK for me.

Reply:
I would possibly prefer to handle this safely memory amount on the `alloc.c` level, but it may require to report different error codes to the config's applier and complicate things so. Anyway, I'm OK with how it is done now, just thoughts.